server/client.go

```diff
@@ -2341,27 +2396,46 @@ func (c *client) pubAllowedFullCheck(subject string, fullCheck bool) bool {
 // Used to mimic client like replies.
 const (
 	replyPrefix    = "_R_."
+	trackSuffix    = ".T"
 	replyPrefixLen = len(replyPrefix)
+	minReplyLen    = 15
 	digits         = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
 	base           = 62
 )
 
 // newServiceReply is used when rewriting replies that cross account boundaries.
 // These will look like _R_.XXXXXXXX.
-func (c *client) newServiceReply() []byte {
+func (c *client) newServiceReply(tracking bool) []byte {
 	// Check to see if we have our own rand yet. Global rand
 	// has contention with lots of clients, etc.
 	if c.in.prand == nil {
 		c.in.prand = rand.New(rand.NewSource(time.Now().UnixNano()))
 	}
 
-	var b = [15]byte{'_', 'R', '_', '.'}
+	var b = [minReplyLen]byte{'_', 'R', '_', '.'}
 	rn := c.in.prand.Int63()
 	for i, l := replyPrefixLen, rn; i < len(b); i++ {
 		b[i] = digits[l%base]
 		l /= base
 	}
-	return b[:]
+	reply := b[:]
+	if tracking {
+		if c.in.shash == nil {
+			c.in.shash = c.srv.sysHash()
```

Comment:
We are not making a copy here, so I don't see the point in doing that (meaning that the slice would be local, but backend would still be "remote"). I would instead make the computation of the sysHash in NewServer only once and therefore would be immutable and could be referenced here. Or if you want cache locality, then it would have to be a copy with local embedded backend.

Reply:
Good point, will fix.